search/services/index/classic.py

```diff
@@ -0,0 +1,93 @@
+
+"""
+Module for translating classic API `Phrase` objects to the
+Elasticsearch DSL and retrieving results.
+"""
+
+from typing import Dict, Callable
+
+from elasticsearch_dsl import Q, Search
+from elasticsearch_dsl.query import QueryString
+
+from ...domain import ClassicAPIQuery, Phrase, Term, Field, Operator
+from .prepare import SEARCH_FIELDS
+
+
+def classic_search(search: Search, query: ClassicAPIQuery) -> Search:
+    """
+    Prepare a :class:`.Search` from a :class:`.ClassicAPIQuery`.
+
+    Parameters
+    ----------
+    search : :class:`.Search`
+        An Elasticsearch search in preparation.
+    query : :class:`.ClassicAPIQuery`
+        An query originating from the Classic API.
+
+    Returns
+    -------
+    :class:`.Search`
+        The passed ES search object, updated with specific query parameters
+        that implement the advanced query.
+
+    """
+    dsl_query = _phrase_to_query(query.phrase)
+
+    return search.query(dsl_query)
+
+
+def _phrase_to_query(phrase: Phrase) -> Q:
+    """
+    Parses a Phrase of a Classic API request into an ES Q object.
+    """
+    # base case - simple term query automatically handled by delegation.
+    if isinstance(phrase[0], Field) and len(phrase) == 2:
+        return _term_to_query(phrase) # type: ignore
+
+    # parse Phrase object an build Q
+    q = Q()
+    current_operator: Operator = Operator.AND
+
+    for token in phrase:
+        if isinstance(token, Operator):
+            current_operator = token
+        elif isinstance(token, tuple):
+            phrase_q: Q = Q()
+            if isinstance(token[0], Operator) or len(token) == 3:
+                phrase_q = _phrase_to_query(token) # type: ignore
```

Comment:
why ignoring type here? seems like we should have everything we need for type checking to be happy here